### ISSUE TITLE
Jointwastesolutions_org.py support for Surrey Heath

### ISSUE THREAD

**Issue by CacheMinimal** [mampfes/hacs_waste_collection_schedule#1131]
Joint Waste Solutions appear to provide waste collections for the Surrey boroughs of Woking, Elmbridge, Mole Valley and Surrey Heath.  The current `jointwastesolutions_org.py` source currently only supports the Woking borough due to the refer URL passed into the various calls and/or headers.

I have edited the script and tested it with an address in Surrey Heath, requiring the four POST/GET calls and two header parameters to be updated from:

`https://asjwsw-wrpwokingmunicipal-live.whitespacews.com/`

to:

`https://asjwsw-wrpsurreyheathmunicipal-live.whitespacews.com/`

Sadly there is no easy way (that I know of) to determine which borough a postcode is in.  Therefore my solution would be to rename `jointwastesolutions_org.py` to `woking_gov_uk.py` and create a second source of `surreyheath_gov_uk.py` with the updated URL and description.  This way the Home Assistant end-user can determine who they pay their Council Tax to.

If this sounds like a good plan, I can possibly have a go at a PR for it?

Thanks for your contributions to making Home Assistant really useful.

Ian.

**Comment by 5ila5** [comment]
You're right it works with Surrey Heath as well but Elmbridge and Mole Valley use a different System and just changing the URL does not seem to work.

it's a pretty fresh source #1124 and not yet part of any release we could add an additional parameter to change the url 
(maybe with Woking as default)

**Comment by 5ila5** [comment]
Did you find any easy way for Elmbridge or Mole Valley? Any improvements for the Source? If not I would just merge my changes suggested in #1132

**Comment by dt215git** [comment]
There's already a source for Elmbridge.
I'm assuming all four councils will migrate to JWS over time as the JWS website currently redirects to the separate councils collection query websites, which seems odd for a consolidated service. That's why I named it JWS rather than Woking, but if you want to change things, I've no problem with that.
Back when the Elmbridge source was written I don't remember seeing any reference to JWS.

**Comment by CacheMinimal** [comment]
I didn't manage to find any easy way for Elmbridge or Mole Valley.  Although it is the same company providing the service, these two boroughs seem to use a different web UI.  It appears there are only two UIs to scrape for the information; the Woking/Surrey Heath and Elmbridge/Mole Valley ones.

Not sure what the best way of handling this is?

**Comment by 5ila5** [comment]
I merged  #1132 which adds support for Surrey Heath Elmbridge got it's own source.

We could write an additional source for Mole Valley. 

If Mole Valley or Elmbridge ever change to the other system/UI they are already supported by the `Jointwastesolutions_org` source and we can just update the EXTRA_INFO variable and add them to the md file

**Comment by bushbrother** [comment]
Hi, it seems the elmbridge pages are stuck in July and are no longer updating on the normal pages, it shows as 300+ days to my next collection due to this! Looks like they may have swapped to this new source?

**Comment by 5ila5** [comment]
I fixed Elmbridge in #2757 

This will be part of the next release or is available now as master (git) version 

install master/git: Developer tools -> actions -> select `update: Install update` -> click `Coose entity` and select WCS -> select `version` and type `master` in the input field -> run the action -> restart HA


<details><summary>Older versions of HACS:</summary>
<p>
install master with HACS version < 2.0: (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
</p>
</details>

**Comment by bushbrother** [comment]
Thank you! Will test on release.

**Comment by markvp** [comment]
Thanks for the original report — this is now fully resolved across all four Joint Waste Solutions councils:

- **Woking Borough Council** and **Surrey Heath Borough Council** are both supported by `jointwastesolutions_org` (use `borough: "woking"` or `borough: "surreyheath"`)
- **Elmbridge Borough Council** has its own dedicated source: `elmbridge_gov_uk`
- **Mole Valley District Council** has its own dedicated source: `mole_valley_gov_uk`

Closing as resolved. If you run into any further issues with these sources, please open a new issue.
